io.openems.edge.io.opendtu/src/io/openems/edge/io/opendtu/inverter/OpendtuImpl.java

```diff
@@ -0,0 +1,268 @@
+
+			BridgeHttp.Endpoint endpoint = new BridgeHttp.Endpoint(this.baseUrl + "/api/limit/config", HttpMethod.POST,
+					BridgeHttp.DEFAULT_CONNECT_TIMEOUT, BridgeHttp.DEFAULT_READ_TIMEOUT, formattedPayload, properties);
+
+			this.httpBridge.request(endpoint).thenAccept(response -> {
+				this.channel(Opendtu.ChannelId.POWER_LIMIT).setNextValue(config.initialPowerLimit());
+				this.channel(Opendtu.ChannelId.POWER_LIMIT_FAULT).setNextValue(false);
+			}).exceptionally(ex -> {
+				this.channel(Opendtu.ChannelId.POWER_LIMIT_FAULT).setNextValue(true);
+				return null;
+			});
+
+			this.isInitialPowerLimitSet = true;
+		}
+	}
+
+	private void processHttpResult(JsonElement result, Throwable error) {
+		this._setSlaveCommunicationFailed(result == null);
+
+		Integer power = null;
+		Integer reactivepower = null;
+		Integer voltage = null;
+		Integer current = null;
+		Integer frequency = null;
+
+		if (error != null) {
+			this.logDebug(this.log, error.getMessage());
+		} else {
+			try {
+				var response = getAsJsonObject(result);
+				var invertersArray = getAsJsonArray(response, "inverters");
+				var firstInverter = invertersArray.get(0).getAsJsonObject();
+				var acData = getAsJsonObject(firstInverter, "AC");
+				var ac0Data = getAsJsonObject(acData, "0");
+
+				var powerObj = getAsJsonObject(ac0Data, "Power");
+				power = round(getAsFloat(powerObj, "v"));
+
+				var reactivePowerObj = getAsJsonObject(ac0Data, "ReactivePower");
+				reactivepower = round(getAsFloat(reactivePowerObj, "v") * 1000);
+
+				var voltageObj = getAsJsonObject(ac0Data, "Voltage");
+				voltage = round(getAsFloat(voltageObj, "v") * 1000);
+
+				var currentObj = getAsJsonObject(ac0Data, "Current");
+				current = round(getAsFloat(currentObj, "v") * 1000);
+
+				var frequencyObj = getAsJsonObject(ac0Data, "Frequency");
+				frequency = getAsInt(frequencyObj, "v") * 1000;
+
+			} catch (OpenemsNamedException e) {
+				this.logDebug(this.log, e.getMessage());
+			}
+		}
+
+		switch (this.phase) {
+		case L1:
+			this._setVoltageL1(voltage);
+			this._setCurrentL1(current);
+			break;
+		case L2:
+			this._setVoltageL2(voltage);
+			this._setCurrentL2(current);
+			break;
+		case L3:
+			this._setVoltageL3(voltage);
+			this._setCurrentL3(current);
+			break;
+		}
+
+		this._setReactivePower(reactivepower);
+		this._setCurrent(current);
+		this._setVoltage(voltage);
+		this._setFrequency(frequency);
+		this._setActivePower(power);
+
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		this.httpBridgeFactory.unget(this.httpBridge);
+		this.httpBridge = null;
+		super.deactivate();
+	}
+
+	@Override
+	public String debugLog() {
+		var b = new StringBuilder();
+		b.append(this.getActivePowerChannel().value().asString());
+		return b.toString();
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		if (!this.isEnabled()) {
+			return;
+		}
+
+		switch (event.getTopic()) {
+		case EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE:
```

Comment:
I guess that the case will never be reached, as it was not mentioned in the event topics in the class annotations

`
@EventTopics({ //
		EdgeEventConstants.TOPIC_CYCLE_BEFORE_PROCESS_IMAGE, //
})
public class OpendtuImpl extends AbstractOpenemsComponent
		implements Opendtu, SinglePhaseMeter, ElectricityMeter, OpenemsComponent, EventHandler, TimedataProvider {
`

For now you could only react on case TOPIC_CYCLE_BEFORE_PROCESS_IMAGE mentioned in the annotation